mlos_bench/mlos_bench/services/remote/azure/azure_auth.py

```diff
@@ -47,12 +51,49 @@ def __init__(self,
         # Register methods that we want to expose to the Environment objects.
         self.register([self.get_access_token])
 
+        if "spClientId" in self.config:
+            check_required_params(
+                self.config, {
+                    "keyVaultName",
+                    "certName",
+                    "spClientId",
+                }
+            )
+
         # This parameter can come from command line as strings, so conversion is needed.
         self._req_interval = float(self.config.get("tokenRequestInterval", self._REQ_INTERVAL))
+        keyvault_name = self.config.get("keyVaultName")
+        cert_name = self.config.get("certName")
+        sp_client_id = self.config.get("spClientId")
```

Comment:
if the key is required, then maybe write
```suggestion
        keyvault_name = self.config["keyVaultName"]
        cert_name = self.config["certName"]
        sp_client_id = self.config["spClientId"]
```

Reply:
It should also be marked in the schema as required then, however, I think we also discussed making this *not* required, so that if someone doesn't have access to create an SP or assign IAM rights to an RG, then they can still use the framework as themselves as a fallback (albeit with some potential token refresh issues).
In that case, we can maybe just issue a warning, and move on.

Reply:
I agree I wanted to make the use of the SP optional in case a user did not have appropriate access to use one. But if they did use the SP, then this group of parameters should be required together. I will move these around to reflect that!

Reply:
Adjusted the schema abit, now if `spClientId` is provided all the other associated params will be required as well.
However if not provided, we fall back to using the user credentials as before.